ipatests/test_webui/test_automember.py

```diff
@@ -267,6 +377,313 @@ def test_rebuild_membership_users(self):
         self.assert_record('dev2')
 
         # Delete group, users and automember rule
-        self.delete('group', [{'pkey': 'devel'}])
-        self.delete('user', [{'pkey': 'dev1'}, {'pkey': 'dev2'}])
-        self.delete('automember', [{'pkey': 'devel'}], facet='searchgroup')
+        self.delete_user_groups('devel')
+        self.delete_users('dev1', 'dev2')
+        self.delete_user_group_rules('devel')
+
+    @screenshot
+    def test_add_multiple_user_group_rules(self):
+        """
+        Test creating and deleting multiple user group rules
+        """
+
+        groups = ['group1', 'group2', 'group3']
```

Comment:
could you consider adding negative tests for adding user group rule and host group rule? like you added for search feature.

Reply:
There are several negative test cases for group rules:
test_add_user_group_rule_with_no_group
test_add_host_group_rule_with_no_group
test_add_user_group_rules_for_same_group
test_add_host_group_rules_for_same_group

Reply:
It is impossible because I need to add group with such name. But it is invalid name. So we have got no group and nothing to test.

Reply:
That is what I would like to see in negative tests i.e invalid name should not be accepted as group name. We can add additional tests for them.

Reply:
But it is not "Groups" tests, it is "Automember" tests.
For "Groups" we already have https://github.com/freeipa/freeipa/blob/master/ipatests/test_webui/test_group.py

Reply:
ahh, sorry. I missed that.